vslib/src/sai_vs_hostintf.cpp

```diff
@@ -36,10 +37,278 @@ typedef struct _hostif_info_t
+
+        if (status != SAI_STATUS_SUCCESS)
+        {
+            continue;
+        }
+
+        if (port_id != attr.value.oid)
+        {
+            // this is not expected port
+            continue;
+        }
+
+        bridge_port_id = bpid;
+
+        SWSS_LOG_DEBUG("found bridge port %s for port %s",
+                sai_serialize_object_id(bridge_port_id).c_str(),
+                sai_serialize_object_id(port_id).c_str());
+
+        attr.id = SAI_BRIDGE_PORT_ATTR_BRIDGE_ID;
+
+        status = vs_generic_get(SAI_OBJECT_TYPE_BRIDGE_PORT, bridge_port_id, 1, &attr);
+
+        if (status != SAI_STATUS_SUCCESS)
+        {
+            break;
+        }
+
+        bridge_id = attr.value.oid;
+
+        SWSS_LOG_DEBUG("found bridge %s for port %s",
+                sai_serialize_object_id(bridge_id).c_str(),
+                sai_serialize_object_id(port_id).c_str());
+
+        attr.id = SAI_BRIDGE_ATTR_TYPE;
+
+        status = vs_generic_get(SAI_OBJECT_TYPE_BRIDGE, bridge_id, 1, &attr);
+
+        if (status != SAI_STATUS_SUCCESS)
+        {
+            break;
+        }
+
+        bridge_type = (sai_fdb_entry_bridge_type_t)attr.value.s32;
+
+        SWSS_LOG_DEBUG("bridge %s type is %d",
+                sai_serialize_object_id(bridge_id).c_str(),
+                bridge_type);
+
+        break;
+    }
+}
+
+void process_packet_for_fdb_event(
+        _In_ const uint8_t *buffer,
+        _In_ size_t size,
+        _In_ const std::shared_ptr<hostif_info_t> &info)
+{
+    MUTEX();
+
+    SWSS_LOG_ENTER();
+
+    uint32_t frametime = (uint32_t)time(NULL);
+
+    /*
+     * We add +2 in case if frame contains 1Q VLAN tag.
+     */
+
+    if (size < (sizeof(ethhdr) + 2))
+    {
+        SWSS_LOG_WARN("ethernet frame is too small: %zu", size);
+        return;
+    }
+
+    const ethhdr *eh = (const ethhdr*)buffer;
+
+    uint16_t proto = htons(eh->h_proto);
+
+    uint16_t vlan_id = DEFAULT_VLAN_NUMBER;
+
+    switch (proto)
+    {
+        case ETH_P_IP:
+        case ETH_P_IPV6:
+        case ETH_P_ARP:
+        case ETH_P_RARP:
+
+            {
+                // IP frame, we need to get vlan if from port
+
+                sai_attribute_t attr;
+
+                attr.id = SAI_PORT_ATTR_PORT_VLAN_ID;
+
+                sai_status_t status = vs_generic_get(SAI_OBJECT_TYPE_PORT, info->portid, 1, &attr);
+
+                if (status != SAI_STATUS_SUCCESS)
+                {
+                    SWSS_LOG_WARN("failed to get port vlan id from port %s",
+                            sai_serialize_object_id(info->portid).c_str());
+                    return;
+                }
+
+                vlan_id = attr.value.u16;
+
+                break;
+            }
+
+        case ETH_P_8021Q:
+
+            {
+                // this is tagged frame, get vlan id from frame
+
+                uint16_t tci = htons(eh->h_proto);
+
+                vlan_id = tci & 0xfff;
+
+                if (vlan_id == 0 || vlan_id == 0xfff)
```

Comment:
>vlan_id == 0  [](start = 20, length = 13)

vlan id can be 0, when it is zero, it mean priority tag packet, should treat as the untagged packet, assigning the pvid to the packet as the vlan_id.

Reply:
pvid  ? port vlan id ?

Reply:
yes, port vlan id.

Reply:
fixdd